### PR TITLE
Add get diagnostic syntax to mysql dialect

### LINKED ISSUE #1148: Add Get Diagnostics syntax to mysql dialect
Add various keywords necessary to support `GET DIAGNOSTICS` functionality to the mysql dialect

### PR DESCRIPTION
**Pull request by silverbullettruck2001**
I also added 2 tests for code that I previously committed and was merged (`function_definer` and `procedure_definer`)

Resolves #1148

### CODE REVIEW

**Review by barrywhart — APPROVED** [review]
Looks good! A couple of observations, but nothing to hold up merging.

**Comment by barrywhart** [comment]
@silverbullettruck2001: Can you fix the `CHANGELOG.md` conflicts? Otherwise, this is good to merge. I'll be "away from keyboard" for a few days. Feel free to tag another maintainer to merge it.

**Review by barrywhart — CHANGES_REQUESTED** [review]
Small issue with the change log

File: CHANGELOG.md
Comment (silverbullettruck2001) [reply CHANGELOG.md:14]: @barrywhart These duplicate lines have been removed.

**Review by barrywhart — APPROVED** [review]

**Pull request merged**
